apiserver/facades/client/controller/controller.go

```diff
@@ -404,9 +409,9 @@ func (c *ControllerAPI) AllModels(ctx context.Context) (params.UserModelList, er
 			},
 		}
 
-		lastConn, err := model.LastModelConnection(c.apiUser)
+		lastConn, err := c.accessService.LastModelLogin(ctx, c.apiUser.Name(), coremodel.UUID(model.UUID()))
 		if err != nil {
```

Comment:
Combining them would change the behavior. In the case where the error is UserNeverAccessedModel then combining them would cause it to be set to lastConn which it is not currently.

I'll change this clause to make it more explicit in setting lastConn to nil because right now its a bit obfuscated.